shared/reducers/chat.js

```diff
@@ -79,6 +79,16 @@ function reducer (state: State = initialState, action: Actions) {
   switch (action.type) {
     case CommonConstants.resetStore:
       return initialState
+    case 'chat:removeOutboxMessage': {
```

Comment:
I think this is incorrect. We handle deleted messages by keeping the delete id in a set and we clear things using our filtering. If you don't add to this set if we get this message somehow (like through loadmessage) it'll reappear